File: Yow.CoD.Finance.Domain/Services/DisburseLoanFundsCommandHandler.cs
using System.Threading.Tasks;
using Yow.CoD.Finance.Domain.Contracts;
using Yow.CoD.Finance.Domain.Model;

namespace Yow.CoD.Finance.Domain.Services
{
    public class DisburseLoanFundsCommandHandler : IHandler<DisburseLoanFundsCommand>
    {
        private readonly IRepository<Loan> _repository;

        public DisburseLoanFundsCommandHandler(IRepository<Loan> repository)
        {
            _repository = repository;
        }

        public async Task Handle(DisburseLoanFundsCommand command)
        {
            var loan = await _repository.Get(command.AggregateId);
            loan.DisburseFunds(command);
            await _repository.Save(loan);
        }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/services/DisburseLoanFundsCommandHandler.java
package com.leecampbell.cod.domain.services;

import com.leecampbell.cod.domain.contracts.*;
import com.leecampbell.cod.domain.model.Loan;

public final class DisburseLoanFundsCommandHandler implements CommandHandler<DisburseLoanFundsCommand, Receipt> {
    private final Repository _repository;

    public DisburseLoanFundsCommandHandler(Repository repository) {
        _repository = repository;
    }

    public Receipt handle(DisburseLoanFundsCommand command) {
        Loan loan = _repository.get(command.getAggregateId());
        loan.disburseFunds(command);
        _repository.save(loan);
        return new CommandReceipt(loan.getId(), loan.getVersion());
    }
}